extensions/ql-vscode/src/query-history-info.ts

```diff
@@ -17,3 +17,16 @@ export function getRawQueryName(item: QueryHistoryInfo): string {
       assertNever(item);
   }
 }
+
+export function getQueryId(item: QueryHistoryInfo): string {
+  switch (item.t) {
+    case 'local':
+      return item.initialInfo.id;
+    case 'remote':
+      return item.queryId;
+    case 'variant-analysis':
+      return item.variantAnalysis.id.toString();
```

Comment:
Appreciate this was kind of my idea but I'm wondering if this is the right id to be using now 🤔 (sorry!)

I think it's okay because both local and remote queries use `nanoid`, but if we had a different item that had an id that is a number (like variant analysis), then this wouldn't be unique enough.

Perhaps we need to prefix with `variant-analysis-${item.variantAnalysis.id}`? Or we could generate one with `nanoid` and store it as part of the history item.

I'm leaning towards the latter (have an id that specifically identifies the history item, and generating that id through `nanoid`). Probably best to get team consensus on this one.

Reply:
Just trying to understand the purpose of this ID. So does this not ID need to be consistent at all between different loadings of the extension, or do we store the generated ID when we serialise the query history?

The definition of `createQueryId` from `remote-queries-manager.ts` says
> Generates a unique id for this query, suitable for determining the storage location for the downloaded query artifacts.

Assuming I understand correctly how this ID is used, the only "benefit" I can see to using `variant-analysis-${item.variantAnalysis.id}` is that we would re-use the same storage location directory for downloaded artifacts. I'm not sure if this is strictly even a benefit or rather a potential bug, and it's certainly not very impactful.

So I also think a `nanoid` is the right option.

Reply:
Thanks! I'll change this to have a new top-level `queryId` property using `nanoid` 👍🏽

Reply:
We also use this id to identify items in the query history (although it looks like we use type + id so it doesn't really have to be unique across all items) e.g. https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/query-history.ts#L722.

Let's go with `nanoid`! Can we call it `historyItemId` or something like that?